tools/autograd/templates/Functions.cpp

```diff
@@ -1,3 +1,10 @@
+// NB: Must be at the top of file to avoid including the deprecated "math.h".
+// https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio
+#ifdef _MSC_VER
+#define _USE_MATH_DEFINES
```

Comment:
...you didn't turn on clang-tidy on Windows, did you?

Reply:
No but I fixed all `x.h` headers to `<cx`> and MSVC has some stupid bug around math.h